simple-podcasting.php

```diff
@@ -19,8 +19,11 @@
 define( 'PODCASTING_PATH', dirname( __FILE__ ) . '/' );
 define( 'PODCASTING_URL', plugin_dir_url( __FILE__ ) );
 define( 'TAXONOMY_NAME', 'podcasting_podcasts' );
+define( 'PODCASTING_TAXONOMY_NAME', 'podcasting_podcasts' );
 define( 'PODCASTING_ITEMS_PER_PAGE', 250 );
 
+trigger_error( 'The constant TAXONOMY_NAME is deprecated and will be removed in future versions. Use PODCASTING_TAXONOMY_NAME instead.', E_USER_DEPRECATED );
+
```

Comment:
The current set up of the plugin doesn't allow site owners to define the constant elsewhere to preempt the value (ie, there is no `if ( defined( ... ) )` check). As such I think we can safely hard deprecate the value and only throw a notice if the old value is defined.

cc @dkotter for a logic check.


```suggestion
define( 'PODCASTING_TAXONOMY_NAME', 'podcasting_podcasts' );
define( 'PODCASTING_ITEMS_PER_PAGE', 250 );

if ( defined( 'TAXONOMY_NAME' ) {
	$message = sprintf(
		/* translators: 1: TAXONOMY_NAME, 2: PODCASTING_TAXONOMY_NAME. */
		__( 'The %1$s constant is no longer supported. Use %2$s instead.', 'simple-podcasting' ),
		'TAXONOMY_NAME',
		'PODCASTING_TAXONOMY_NAME'
	);
	trigger_error( wp_strip_all_tags( $message ), E_USER_DEPRECATED );
}
```

Reply:
This looks good to me, though as mentioned, we didn't really support this value being overridden in the first place so not sure there's a scenario where this deprecation notice would even be used. Doesn't hurt to have it in place though, so I'm fine with this change

Reply:
@peterwilsoncc 
`TAXONOMY_NAME` has no prefix specific to **Simple Podcasting** plugin. What if any other script outside **Simple Podcasting** plugin uses this constant for their own purpose later? In that case showing deprecation notice  might be confusing.

Reply:
@dkotter @jayedul That makes sense, let's just replace the constant without a deprecation notice.